extends BaseGuiComponent

@export var worker: WorkComponent

@onready
var _progress_bar: TextureProgressBar = $PanelContainer/MarginContainer/VBoxContainer/ProgressBar


func _ready():
	assert(worker, "missing work component")

	panels = [
		$PanelContainer/MarginContainer/VBoxContainer/HBoxContainer/InputPanel,
		$PanelContainer/MarginContainer/VBoxContainer/HBoxContainer/OutputPanel
	]

	panels[worker.INPUT_SLOT].item_filter = worker.input_name
	panels[worker.OUTPUT_SLOT].item_filter = worker.output_name

	_progress_bar.value = 0
	_progress_bar.max_value = 1

	worker.update_progress.connect(_on_update_progress)
	worker._slots[worker.INPUT_SLOT].bind(panels[worker.INPUT_SLOT])
	worker._slots[worker.OUTPUT_SLOT].bind(panels[worker.OUTPUT_SLOT])


func _on_update_progress(percent: float) -> void:
	_progress_bar.value = percent
